datasets/cz/national_sanctions/crawler.py

```diff
@@ -0,0 +1,158 @@
+from typing import Generator, cast, List
+from zavod import Context, helpers as h
+import openpyxl
+from openpyxl import load_workbook
+from pantomime.types import XLSX
+from normality import stringify, slugify
+from datetime import datetime
+
+CZECH_MONTH_MAPPING = {
+    "ledna": "January",
+    "února": "February",
+    "března": "March",
+    "dubna": "April",
+    "května": "May",
+    "června": "June",
+    "července": "July",
+    "srpna": "August",
+    "září": "September",
+    "října": "October",
+    "listopadu": "November",
+    "prosince": "December",
+}
+
+
+def translate_date(date: str) -> str:
+    """
+    Translates Czech month names to English month names in a given date string.
+    """
+    for czech_month, eng_month in CZECH_MONTH_MAPPING.items():
+        date = date.replace(czech_month, eng_month)
+    return date
+
+
+def parse_sheet(
+    sheet: openpyxl.worksheet.worksheet.Worksheet,
+) -> Generator[dict, None, None]:
+    headers = None
+    for row in sheet:
+        cells = [c.value for c in row]
+        if headers is None:
+            headers = []
+            for idx, cell in enumerate(cells):
+                if cell is None:
+                    cell = f"column_{idx}"
+                headers.append(slugify(cell, "_").lower())
+            continue
+
+        record = {}
+        for header, value in zip(headers, cells):
+            if isinstance(value, datetime):
+                value = value.date()
+            record[header] = stringify(value)
+        if len(record) == 0:
+            continue
+        yield record
+
+
+def crawl_item(input_dict: dict, context: Context):
+    # Jméno fyzické osoby -> First name of the natural person
+    first_name = input_dict.pop("jmeno_fyzicke_osoby")
+
+    # Příjmení fyzické osoby / Název právnické osoby / Označení nebo název entity -> Last name of the natural person / Name of the legal entity / Designation or name of the entity
+    last_name = input_dict.pop(
+        "prijmeni_fyzicke_osoby_nazev_pravnicke_osoby_oznaceni_nebo_nazev_entity"
+    )
+
+    # Datum narození fyzické osoby -> Date of birth of the natural person
+    birth_date = input_dict.pop("datum_narozeni_fyzicke_osoby")
+
+    # Státní příslušnost fyzické osoby / sídlo právnické osoby -> Nationality of the natural person / registered office of the legal entity
+    res = context.lookup(
+        "countries_names",
+        input_dict.pop("statni_prislusnost_fyzicke_osoby_sidlo_pravnicke_osoby"),
+    )
+    if res:
+        countries = cast("List[str]", res.names)
+    else:
+        context.log.warning("Country not identified", text=res)
+        countries = None
+
+    if first_name is None:
+        entity = context.make("LegalEntity")
+        entity.id = context.make_slug(last_name)
```

Comment:
We do need a more detailed ID here. Do we have any sort of extra key material we can use?

Reply:
For Legal entities, the only other information that is not related to the sanction is the country, we can use that. What do you think?